core/src/main/java/greencity/webcontroller/ManagementAchievementController.java

```diff
@@ -0,0 +1,125 @@
+@RequestMapping("/management/achievement")
+public class ManagementAchievementController {
+    private final AchievementService achievementService;
+    private final AchievementCategoryService achievementCategoryService;
+    private final LanguageService languageService;
+
+    /**
+     * Method that returns management page with all {@link AchievementVO}.
+     *
+     * @param model    Model that will be configured and returned to user.
+     * @param pageable {@link Pageable}.
+     * @return View template path {@link String}.
+     * @author Orest Mamchuk
+     */
+    @GetMapping
+    public String getAllAchievement(@RequestParam(required = false, name = "query") String query, Pageable pageable,
+        Model model) {
+        Pageable paging = PageRequest.of(pageable.getPageNumber(), pageable.getPageSize(), Sort.by("id").descending());
+        PageableAdvancedDto<AchievementVO> allAchievements = query == null || query.isEmpty()
+            ? achievementService.findAll(paging)
+            : achievementService.searchAchievementBy(paging, query);
+        model.addAttribute("pageable", allAchievements);
+        model.addAttribute("categoryList", achievementCategoryService.findAll());
+        model.addAttribute("languages", languageService.getAllLanguages());
+        return "core/management_achievement";
+    }
+
+    /**
+     * Method for creating {@link AchievementVO}.
+     *
+     * @param achievementPostDto dto for {@link AchievementVO} entity.
+     * @return {@link GenericResponseDto} with of operation and errors fields.
+     * @author Orest Mamchuk
+     */
+    @PostMapping
+    @ResponseBody
+    public GenericResponseDto saveAchievement(@Valid @RequestBody AchievementPostDto achievementPostDto,
+        BindingResult bindingResult) {
+        if (!bindingResult.hasErrors()) {
+            achievementService.save(achievementPostDto);
+        }
+        return buildGenericResponseDto(bindingResult);
+    }
+
+    /**
+     * Method which deteles {@link AchievementVO} by given id.
+     *
+     * @param id of {@link AchievementVO}
+     * @return {@link ResponseEntity}
+     * @author Orest Mamchuk
+     */
+    @DeleteMapping("delete/{id}")
+    public ResponseEntity<Long> deleteAdviceById(@PathVariable Long id) {
+        return ResponseEntity.status(HttpStatus.OK)
+            .body(achievementService.delete(id));
+    }
+
+    /**
+     * Method for deleting {@link AchievementVO} by given id.
+     *
+     * @param listId list of IDs.
+     * @return {@link ResponseEntity}
+     * @author Orest Mamchuk
+     */
+    @DeleteMapping("/deleteAll")
+    public ResponseEntity<List<Long>> deleteAll(@RequestBody List<Long> listId) {
+        achievementService.deleteAll(listId);
+        return ResponseEntity.status(HttpStatus.OK).body(listId);
+    }
+
+    /**
+     * Method for getting econews by id.
+     *
+     * @param id of Eco New
+     * @return {@link AchievementVO} instance.
+     */
+    @GetMapping("/find")
```

Comment:
@RequestParam("id") Long id - move this to path parama as {id}, and remove 'find' from relative path

Reply:
Done